File: Array_problems/RangeSumQuery.java

// Problem link - https://leetcode.com/problems/range-sum-query-immutable/description/

package Array_problems;

class NumArray {
    private int[] nums;
    public NumArray(int[] nums) {
        this.nums = nums;
    }
    
    public int sumRange(int left, int right) {
        int rangeSum = 0;
        for(int i=left; i<=right; i++){
            rangeSum += nums[i];
        }
        return rangeSum;
    }
}

public class RangeSumQuery {
    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        NumArray numArray = new NumArray(nums);

        // Test the sumRange method
        System.out.println(numArray.sumRange(0, 2)); // Output: 6 (1 + 2 + 3)
        System.out.println(numArray.sumRange(1, 3)); // Output: 9 (2 + 3 + 4)
        System.out.println(numArray.sumRange(2, 4)); // Output: 12 (3 + 4 + 5)
    }
}

File: Array_problems/RangeSumQuery.cpp

// Problem link - https://leetcode.com/problems/range-sum-query-immutable/description/

#include<bits/stdc++.h>
using namespace std;

class NumArray {
private:
    vector<int> nums;
public:
    NumArray(vector<int>& nums) {
        this->nums = nums;
    }
    
    int sumRange(int left, int right) {
        int sum = 0;
        for(int i=left; i<=right; i++){
            sum += nums[i];
        }
        return sum;
    }
};

int main(){
    vector<int> nums = {1, 2, 3, 4, 5};
    NumArray numArray(nums);

    // Test the sumRange method
    cout<<numArray.sumRange(0, 2)<<endl; // Output: 6 (1 + 2 + 3)
    cout<<numArray.sumRange(1, 3)<<endl; // Output: 9 (2 + 3 + 4)
    cout<<numArray.sumRange(2, 4)<<endl; // Output: 12 (3 + 4 + 5)
    return 0;
}